src/services/blocks/BlocksService.ts

```diff
@@ -644,6 +644,20 @@ export class BlocksService extends AbstractService {
 		}
 	}
 
+	/**
+	 * Checks to see if the current chain has the session module, then retrieve all
+	 * validators.
+	 *
+	 * @param historicApi ApiDecoration to use for the query
+	 */
+	private async fetchValidators(
+		historicApi: ApiDecoration<'promise'>
+	): Promise<Vec<AccountId32>> {
+		return historicApi.query.session
+			? historicApi.query.session.validators()
+			: ([] as unknown as Vec<AccountId32>);
```

Comment:
I think this was brought up before but i forgot; why do we need to cast to unknown before Vec?

Reply:
So this comes down to the typescript compiler. Basically, the functions expects `Vec<AccountId32>` to be returned, and when we return [] its of type Array<any>. Now the compiler sees `Vec<AccountId32>` and notices there is no straight type conversion from `Vec<AccountId32>` -> `Array<string>`, therefore it throws a type conversion error (Even though we know the exact output and input for the values). To get around type conversion issue we have to set it to unknown so the typescript compiler sees that the type is convertable.